### PR TITLE
[BEAM-1340] Add __all__ tags to modules in package apache_beam/transforms

### PR DESCRIPTION
**Pull request by charlesccychen**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @robertwb

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -53,6 +53,37 @@
 from apache_beam.options.pipeline_options import TypeOptions
 
 
+__all__ = [
+    'DoFnContext',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: DoFnContext should not be public.

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -53,6 +53,37 @@
...
+    'DoFnProcessContext',
+    'DoFn',
+    'CallableWrapperDoFn',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: The CallableWrapper* classes shouldn't be public.

File: sdks/python/apache_beam/transforms/core.py [hunk 3]
```diff
@@ -53,6 +53,37 @@
...
+    'PartitionFn',
+    'CallableWrapperPartitionFn',
+    'ParDo',
+    'FlatMap',
+    'Map',
+    'Filter',
+    'CombineGlobally',
+    'CombinePerKey',
+    'CombineValues',
+    'CombineValuesDoFn',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: CombineValuesDoFn should not be public.

File: sdks/python/apache_beam/transforms/core.py [hunk 4]
```diff
@@ -53,6 +53,37 @@
...
+    'GroupByKey',
+    'ReifyWindows',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: ReifyWindows, GroupAlsoByWIndows, and GroupByKeyOnly should not be public.

File: sdks/python/apache_beam/transforms/core.py [hunk 5]
```diff
@@ -53,6 +53,37 @@
...
+    'GroupAlsoByWindow',
+    'GroupByKeyOnly',
+    'Partition',
+    'ApplyPartitionFnFn',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: ApplyPartitonFn should not be public.

File: sdks/python/apache_beam/transforms/core.py [hunk 6]
```diff
@@ -53,6 +53,37 @@
...
+    'Windowing',
+    'WindowInto',
+    'WindowIntoFn',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: WindowIntoFn should not be public.

File: sdks/python/apache_beam/transforms/cy_combiners.py [hunk 7]
```diff
@@ -22,6 +22,33 @@
 from apache_beam.transforms import core
 
 
+__all__ = [
+    'AccumulatorCombineFn',
+    'CountAccumulator',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/cy_combiners.py]: I don't think any of these accumulator types should be public. Actually, to be safe, just make this entire module private.

File: sdks/python/apache_beam/transforms/ptransform.py [hunk 8]
```diff
@@ -60,6 +60,17 @@ class and wrapper class that allows lambda functions to be used as
 from apache_beam.utils import urns
 
 
+__all__ = [
+    'GetPValues',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/ptransform.py]: The only exported things from here should be PTransform and possibly label_from_callable. Maybe ptransform_fn, but mark it clearly as experimental (CloudML had some closure issues that were never fully resolved). The rest are internal implementation details.

File: sdks/python/apache_beam/transforms/sideinputs.py [hunk 9]
```diff
@@ -27,6 +27,12 @@
 from apache_beam.transforms import window
 
 
+__all__ = [
+    'default_window_mapping_fn',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/sideinputs.py]: Don't expose anything from this module (yet).

File: sdks/python/apache_beam/transforms/timeutil.py [hunk 10]
```diff
@@ -23,6 +23,17 @@
 from abc import abstractmethod
 
 
+__all__ = [
+    'TimeDomain',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/timeutil.py]: Only TimeDomain should be exported from this module.

File: sdks/python/apache_beam/transforms/trigger.py [hunk 11]
```diff
@@ -38,6 +38,31 @@
...
+    'TriggerFn',
+    'DefaultTrigger',
+    'AfterWatermark',
+    'AfterCount',
+    'Repeatedly',
+    'AfterAny',
+    'AfterAll',
+    'AfterEach',
+    'OrFinally',
+    'TriggerContext',
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/trigger.py]: TriggerContext and below are an implementation details.

File: sdks/python/apache_beam/transforms/window.py [hunk 12]
```diff
@@ -65,6 +65,21 @@
...
+    'BoundedWindow',
+    'IntervalWindow',
+    'TimestampedValue',
+    'GlobalWindow',
+    'NonMergingWindowFn',
+    'GlobalWindows',
+    'FixedWindows',
+    'SlidingWindows',
+    'Sessions',
+    ]
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/window.py]: @aaltay Was the intent to export WindowedValue from here rather than from utils?

Comment (robertwb) [inline sdks/python/apache_beam/transforms/window.py]: Or should we keep it Experimental for now (it's not an exposed concept in Java).

Comment (aaltay) [inline sdks/python/apache_beam/transforms/window.py]: I think we should move `WindowedValue` out of `util` to here and also mark it as experimental. (cc: @sb2nov)

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/window.py]: Marked it is experimental in https://github.com/apache/beam/pull/3087

**Comment by charlesccychen** [comment]
Thanks, PTAL.

File: sdks/python/apache_beam/transforms/ptransform.py [hunk 13]
```diff
@@ -639,6 +646,8 @@ def default_label(self):
 def ptransform_fn(fn):
   """A decorator for a function-based PTransform.
 
+  For internal use only; no backwards-compatibility guarantees.
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/ptransform.py]: Not for internal use, just note that it's expiremental (and, yes, not backwards compatible).

File: sdks/python/apache_beam/transforms/sideinputs.py [hunk 14]
```diff
@@ -27,6 +29,9 @@
 from apache_beam.transforms import window
 
 
+__all__ = []
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/sideinputs.py]: This may cause lint errors. Remove.

**Comment by robertwb** [comment]
R: @sb2nov

**Comment by charlesccychen** [comment]
Thanks, PTAL.

**Comment by robertwb** [comment]
LGTM

**Comment by robertwb** [comment]
There were a couple of trivial test failures (e.g. due to no beam.GroupByKeyOnly). Fixing as part of the merge.

**Pull request closed without merging**